### PR TITLE
Fixed names in masterData.json

### PR DESCRIPTION
**Pull request by AndreiSoroka**
Trim names:

```
"Macao SAR, China                              " => "Macao SAR, China"
"Micronesia, Fed. Sts.                       " => "Micronesia, Fed. Sts."
"Saint Martin","St. Martin (French part)                      "  => "Saint Martin","St. Martin (French part)"
```

### CODE REVIEW

**Comment by ohitsdaniel** [comment]
Hi @AndreiSoroka ! 
First of all, thank you for your contribution and opening this PR. :) 

As these are manual changes in the masterData.json file, they would get overridden whenever someone runs the Makefile. Also, there could be other non-trimmed names that we missed. What do you think about adapting `/lib/data/cleaner.js` to .trim() all name entries inserted into the names array of each country instead? 

(see
  https://github.com/ohitsdaniel/country./blob/master/lib/data/cleaner.js#L75, 
  https://github.com/ohitsdaniel/country./blob/master/lib/data/cleaner.js#L169. 
  https://github.com/ohitsdaniel/country./blob/master/lib/data/cleaner.js#L226,  
  https://github.com/ohitsdaniel/country./blob/master/lib/data/cleaner.js#L313
)

**Comment by AndreiSoroka** [comment]
P.s. offtopic
how I usage the package:
```javascript
import countryInfoData from 'countryinfo/lib/data/masterData.json';
import EuMembers from 'is-eu-member';

// generate map for fast search
const countriesMap = {
  ...Object.values(countryInfoData.code.ISO2).reduce((result, value) => {
    // eslint-disable-next-line no-param-reassign
    result[value.ISO1] = value;
    // eslint-disable-next-line no-param-reassign
    result[value.ISO2] = value;
    // eslint-disable-next-line no-param-reassign
    result[value.ISO3] = value;

    for (const name of [...value.names.en, ...(value.names?.ru || [])]) {
      // eslint-disable-next-line no-param-reassign
      result[name.toUpperCase()] = value;
    }
    return result;
  }, {}),
};

export default function getCountry(name) {
  const data = countriesMap[name.trim().toUpperCase()];

  if (!data) {
    return { isEuMember: false };
  }

  return {
    ...data,
    isEuMember: EuMembers.isEuMember(data?.ISO2 || '') || false,
  };
}
```
because the package can find only 'Germany', but can't find 'germany', 'GERMANY'. And in `find()` a little bit expensive use `toUpperCase each iteration`.

Maybe is updating keys to `toUpperCase` in `masterData.json`?

**Comment by AndreiSoroka** [comment]
I added .editorconfig file because for default I usage space (not tab). Editorconfig auto-configure IDE for project.

File: TODO.md [hunk 1]
```diff
@@ -0,0 +1,2 @@
+Will fix double space inside. Example:
+- `Birmania  Myanmar` to `Birmania Myanmar`
```
Comment (ohitsdaniel) [inline TODO.md:2]: Let's open an issue in the repo fot this instead of introducing a second place to track ToDos. :)

File: lib/data/cleaner.js [hunk 2]
```diff
@@ -12,74 +13,72 @@ const curData = require('country-data').currencies;
...
 
 	for (let i = 1; i < 9; i++) {
-		let name = element['name' + i];
-
-		if (typeof name !== 'undefined') {
-			if (name !== '') {
-				newElement.names['en'].push(name);
-			}
+		const name = element['name' + i];
+		if (name) {
```
Comment (ohitsdaniel) [inline lib/data/cleaner.js:73]: Can we check that the name is not empty here? :)

File: README.md [hunk 3]
```diff
@@ -167,3 +167,9 @@ To import the data into the library, run `make` in the root directory. This recr
 
 ## I don't need a JS library, I just need the data.
 Run `make` in the root directory of the project and copy the `/lib/data/masterData.json` file. This file contains all the mentioned data.
+
+## Build / Test / Update
+In package json:
+- `npm test` - run tests
+- `npm run cleaner` - pipe out the master data
+- `npm run cleaner:update` - update master data
```
Comment (ohitsdaniel) [inline README.md:175]: Regarding this addition: Let's remove the Makefile then and only offer one option in the form of `npm run cleaner`. We then also no longer need the :update flag, as I see no benefit of printing out the masterData to console. Any use case that I might be missing here?

**Review by ohitsdaniel — COMMENTED** [review]
Rest looks great! Thanks for all the effort you've put into this.

Regarding the name lookup, couldn't we do something along the lines of:

```js
// helpers 
function findByName(name, language) {
        let lowercasedName = name.toLowercase();

	let key = Object.keys(data.code.ISO2).find((element) => {
		if (typeof data.code.ISO2[element].names[language] === 'undefined') {
			throw 'INVALIDLANGUAGE: ' + language + 'not supported.'
		}

		return data.code
		    .ISO2[element]
		    .names[language]
		    .map((name) => { name.toLowerCase() })
		    .includes(lowercasedName);
	});

	return data.code.ISO2[key];
}
```

and export it from countrysynonyms.js?

File: lib/data/cleaner.js (hunk 2 above)
Comment (AndreiSoroka) [reply lib/data/cleaner.js:73]: yep
```
'' == false
undefined == false
```
https://javascript.info/type-conversions

File: README.md (hunk 3 above)
Comment (AndreiSoroka) [reply README.md]: I didn't know, why need a console. And just stay that functional.

Removed

File: TODO.md (hunk 1 above)
Comment (AndreiSoroka) [reply TODO.md]: https://github.com/ohitsdaniel/country./issues/4

**Comment by AndreiSoroka** [comment]
> Regarding the name lookup, couldn't we do something along the lines of:
https://github.com/ohitsdaniel/country./issues/5

I think one MR should be fixed one problem... maybe in the future

**Review by ohitsdaniel — APPROVED** [review]
Good work! 🙌 I'll release a new version of this package once this PR is merged.

**Comment by AndreiSoroka** [comment]
Cool! Can you update https://www.npmjs.com/package/countryinfo too?

**Comment by ohitsdaniel** [comment]
Yes, first thing tomorrow morning :)

**Comment by ohitsdaniel** [comment]
Did not find the time to do it today, sorry! Will look into it tomorrow.

**Comment by ohitsdaniel** [comment]
@AndreiSoroka I bumped the dependency versions and released v1.0.4 (old dependency version) and v1.0.5 (bumped dependency version). Thank you for your contribution! :)

**Pull request merged**